### PR TITLE
Problem (fix #471): the default GasPerCollection value is arbitrary

### LINKED ISSUE #471: Problem: the default GasPerCollection value is arbitrary
### Parameters

- Size of plan: `a`
  - `5000(max description) + 140(max title) = 5140` (ignore other fields)
- Size of subscription: `b`
  - `int64 * 5 + int32 + bech32 address = 40 + 4 + 42 = 86`
- RoundUp cpu cost: `c`
  - `1000`, (arbitrary)
- `read_flat/write_flat/read_per_byte/write_per_byte/delete/iter_next`, from [`KVGasConfig`](https://github.com/cosmos/cosmos-sdk/blob/master/store/types/gas.go#L165)

### Create subscription

- RoundUp cpu cost, `c`

### Begin block

- iteration, `iter_next`.
- load subscription, `read_flat + read_per_byte * b`
- load plan, `read_flat + read_per_byte * a`
- coin transfer (ignore the bytes to keep it simple)
  - load account meta, `read_flat`
  - for each denom:
    - load balance, `read_flat`
    - set balance, `write_flat`
- update next collection time
  - RoundUp cpu cost, `c`
  - delete in queue, `delete`
  - insert in queue, `write_flat`
  - Store subscription, `write_flat + write_per_byte * b`

### Sum up

```
read_flat * 4 + 
write_flat * 3 + 
read_per_byte * (a + b) + 
write_per_byte * b +
iter_next +
delete +
c * 2
= 31288
```

### Other
Ignored other aspects for simplicity
- the bytes in kv operation in coin transfer.
- failed collection and retries.

### PR DESCRIPTION
**Pull request by yihuang**
Solution:
- estimate one based on default gas config

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

**Review by tomtau — APPROVED** [review]
makes sense -- given the failed collections and retries aren't "counted in" , should the default "failure tolerance" be decreased?

**Comment by yihuang** [comment]
> makes sense -- given the failed collections and retries aren't "counted in" , should the default "failure tolerance" be decreased?

currently, it retries for every block, I guess it don't make much sense this way, better to keep it low for now, 3?

File: x/subscription/spec/05_params.md [hunk 1]
```diff
@@ -6,12 +6,66 @@ order: 5
 
 The subscription module contains the following parameters:
 
-| Key                 | Type | Example |
+| Key                 | Type | Default |
 | ------------------- | ---- | ------- |
-| GasPerCollection    | int  | "10000" |
+| GasPerCollection    | int  | 31288   |
 | SubscriptionEnabled | bool | true    |
 | FailureTolerance    | int  | 30      |
```
Comment (tomtau) [inline x/subscription/spec/05_params.md:13]:
```suggestion
| FailureTolerance    | int  | 3      |
```

File: x/subscription/keeper/keeper.go [hunk 2]
```diff
@@ -395,6 +394,8 @@ func (k Keeper) TryCollect(ctx sdk.Context, subscription types.Subscription) boo
 		subscription.PaymentFailures++
 		if subscription.PaymentFailures > failureTolerance {
 			k.RemoveSubscription(ctx, subscription)
+		} else {
+			ctx.KVStore(k.storeKey).Set(types.SubscriptionKey(subscription.SubscriptionId), k.MustMarshalSubscription(subscription))
```
Comment (JayT106) [inline x/subscription/keeper/keeper.go:398]: formatting?

**Review by JayT106 — APPROVED** [review]

**Comment by tomtau** [comment]
state sync test fails for some reason: https://github.com/crypto-org-chain/chain-main/pull/476/checks?check_run_id=2329424538#step:5:752

**Pull request merged**